### DISCUSSION TITLE
Include examples when leaflet is used as a module?

### DISCUSSION THREAD

**Discussion by ajmas** [Leaflet/Leaflet#8609] [General]
I was looking through the documentation and I couldn't see any examples where leaflet is used as a module. They all assume it is being loaded directly into HTML with 'L' being in the global space.

I wondering whether there should be at least one example where leaflet is being used via npm?

This would also help address how the package should be imported. For example, I am seeing some people suggest both:

```
import * as L from 'leaflet';
import L  from 'leaflet';
```

and then there is what seems to be a [breaking change](https://github.com/Leaflet/Leaflet/discussions/7729#discussioncomment-1576995), for which I am not seeing any of the examples seemingly address.

The idea here is two-fold:

  - provide documentation
  - establish whether the packaging causes breakages when used a dependency this way
 
A further step could even to be to include references to wrapper packages, that allow using with Vue, Angular and React?

**Comment by jonkoops** [comment]
The idea for version 2 of Leaflet is that we're stepping away from using globals entirely. Instead we will be providing a ESM-native implementation, which also should work in the browser without extra steps.

This naturally also means we're updating the documentation with best practices in this regard.

**Comment by ajmas** [comment]
@jonkoops thanks for the info.

BTW would it be worth making use of CodePen, such as Quasar does for its docs. As an example: https://quasar.dev/vue-components/color-picker
